Owncloud iOs Client/Login/Login/UniversalLoginViewController.swift

```diff
@@ -933,24 +923,31 @@ public enum TextfieldType: String {
                     
                 } else {
 
+                    //1.Generate correct user
+                    if self.loginMode == .create {
```

Comment:
hmm i don't really like that. 🤔

I think this should be coded into a switch, Swift switches are really usefull and powerfull.
You could change the _forceAccountMigration_ global variable into another case in the enumetarion. This way you could use a switch.

Also I think you could extract the common things like setting the user url, username, ssl,... into a function and call it when you need it. If you do this you could pass as parameter all the new values and also the user (maybe marking the user as _inout_). This will create a more readable, decoupled and testable code.

I think worth the little effort needed to accomplish this change.

Reply:
If looks cleaner with switch I replaced it, but the forAccountMigration for this round will not be refactor as another mode.